include/deal.II/base/tensor_base.h

```diff
@@ -430,79 +464,85 @@ class Tensor<1,dim,Number>
    * Number.
    */
   template <typename OtherNumber>
-  Tensor<1,dim,Number> &operator = (const Tensor<1,dim,OtherNumber> &rhs);
+  Tensor &operator = (const Tensor<rank_,dim,OtherNumber> &rhs);
 
   /**
    * This operator assigns a scalar to a tensor. To avoid confusion with what
    * exactly it means to assign a scalar value to a tensor, zero is the only
    * value allowed for <tt>d</tt>, allowing the intuitive notation
    * <tt>t=0</tt> to reset all elements of the tensor to zero.
+   *
+   * @relates EnableIfScalar
    */
-  Tensor<1,dim,Number> &operator = (const Number d);
+  template <typename OtherNumber,
+            typename = typename EnableIfScalar<OtherNumber>::type>
+  Tensor<rank_,dim,Number> &operator = (const OtherNumber d);
```

Comment:
The big problem is that with that it is perfectly valid to write
```
Tensor<1,dim> foo = Tensor<4,dim>();
```
I will come up with something that still allows
```
Tensor<rank,dim,complex<double>> foo = 0.;
```
to be written.